plugins/c8y_firmware_plugin/src/main.rs

```diff
@@ -104,24 +109,30 @@ async fn main() -> Result<(), FirmwareManagementError> {
     let http_address = tedge_config.query(HttpBindAddressSetting)?.to_string();
     let local_http_host = format!("{}:{}", http_address, http_port);
 
-    let tmp_dir = tedge_config.query(TmpPathSetting)?.into();
+    let tmp_dir: PathBuf = tedge_config.query(TmpPathSetting)?.into();
     let timeout_sec = Duration::from_secs(
         tedge_config
             .query(FirmwareChildUpdateTimeoutSetting)?
             .into(),
     );
 
+    let (req_sndr, req_rcvr) = mpsc::channel::<DownloadRequest>(1);
+    let (res_sndr, res_rcvr) = mpsc::channel::<DownloadResult>(1);
```

Comment:
I did that intentionally to limit the number of active downloads to only one. In a follow-up PR, we'll enable parallel downloads with this level of parallelism being a configurable parameter.

Reply:
Increased the the buffer size so that the issue @didier-wenzek highlighted does not occur. But the `DownloadManager` still consumes only one download request at a time and hence there will only be one download happening at a time.